frontend/src/components/CreateEvent/CreateEvent.jsx

```diff
@@ -1,10 +1,90 @@
+    const [startTime, setStartTime] = useState('');
+    const [endDate, setEndDate] = useState('');
+    const [endTime, setEndTime] = useState('');
+    const [location, setLocation] = useState('');
+    const [allDay, setAllDay] = useState(false);
+    const [participants, setParticipants] = useState('');
+
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        const eventData ={
+            title: title,
+            description: description,
+            startDate: startDate,
+            startTime: startTime,
+            endDate: endDate,
+            endTime: endTime,
+            location: location,
+            allDay: allDay,
+            participants: participants
+        };
+        onSubmit(eventData);
+    }
+
+    const handleAllDayClicked = (e) => {
+        setAllDay(e.target.checked);
+        if (!allDay){
+            setStartTime("00:00");
+            setEndTime("23:59");
+            setEndDate(startDate);
+        }
+    }
     return (
-        <>
-            This is the Calendar Event Creation / Editing Page
-        </>
+        <div className="modal-overlay">
+            <div className="modal-content">
+                <form onSubmit={handleSubmit}>
+                    <h1>Create a New Event</h1>
+                    <p>Event Title</p>
+                    <input type="text" value={title} onChange={(e) => setTitle(e.target.value)} placeholder="Event Title" required/>
+                    <p>Event Description</p>
+                    <textarea name="description" id="notes" rows="5" value={description} onChange={(e) => setDescription(e.target.value)}/>
+                    <div className="date-time-container">
+                        <div className="selector-obj">
+                            <p>Start Date</p>
+                            <input type="date" placeholder="mm/dd/yyyy" value={startDate} onChange={(e) => setStartDate(e.target.value)} required/>
+                        </div>
+                        {!allDay && <>
+                            <div className="selector-obj">
+                                <p>Start Time</p>
+                                <input type="time" value={startTime} onChange={(e) => setStartTime(e.target.value)} required/>
+                            </div>
+                        </>}
+                    </div>
+                    <div className="date-time-container">
+                        <div className="selector-obj">
+                            <p>End Date</p>
+                            <input type="date" placeholder="mm/dd/yyyy" value={endDate} onChange={(e) => setEndDate(e.target.value)} required/>
+                        </div>
+                        {!allDay && <>
+                            <div className="selector-obj">
+                                <p>End Time</p>
+                                <input type="time" value={endTime} onChange={(e) => setEndTime(e.target.value)} required/>
```

Comment:
You should have some edge-case handling here where the end time has to be after the start time